code/datums/components/squeak.dm

```diff
@@ -0,0 +1,88 @@
+// Squeak component ported over from tg
+// Note to future coders: I decided to remove the special case for squeak cooldown on usage and instead opt for a universal 20 ticks cooldown to avoid squeak spam
+
+/datum/component/squeak
+	var/static/list/default_squeak_sounds = list('sound/items/toysqueak1.ogg'=1, 'sound/items/toysqueak2.ogg'=1, 'sound/items/toysqueak3.ogg'=1)
+	var/list/override_squeak_sounds
+	var/squeak_chance = 100
+	var/volume = 30
+
+	// This is so shoes don't squeak every step
+	var/steps = 0
+	var/step_delay = 1
+
+	// This is to stop squeak spam from inhand usage
+	var/last_use = 0
+	var/use_delay = 20
+
+/datum/component/squeak/Initialize(custom_sounds, volume_override, chance_override, step_delay_override, use_delay_override)
+	if(!isatom(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, list(COMSIG_ATOM_ENTERED, COMSIG_ATOM_BLOB_ACT, COMSIG_ATOM_HULK_ATTACK, COMSIG_PARENT_ATTACKBY), .proc/play_squeak)
```

Comment:
for `COMSIG_PARENT_ATTACKBY`, make sure it exists for the snowflake interaction of bags on ore.